core/src/main/java/org/elasticsearch/rest/action/support/RestActions.java

```diff
@@ -100,7 +109,16 @@ public static QuerySourceBuilder parseQuerySource(RestRequest request) {
         if (defaultOperator != null) {
             queryBuilder.defaultOperator(Operator.fromString(defaultOperator));
         }
-        return new QuerySourceBuilder().setQuery(queryBuilder);
+        return queryBuilder;
+    }
+
+    public static SearchSourceBuilder getRestSearchSource(BytesReference sourceBytes, IndicesQueriesRegistry queryRegistry)
+            throws IOException {
+        XContentParser parser = XContentFactory.xContent(sourceBytes).createParser(sourceBytes);
+        QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
```

Comment:
I think the current behaviour is trappy for devs then. We should make the constructor have a parameter for parseFieldMatcher so it is obvious it needs to be set. wdyt?